crates/ruff_cli/src/lib.rs

```diff
@@ -236,17 +237,27 @@ pub fn check(args: CheckCommand, log_level: LogLevel) -> Result<ExitStatus> {
 
     // Fix rules are as follows:
     // - By default, generate all fixes, but don't apply them to the filesystem.
-    // - If `--fix` or `--fix-only` is set, always apply fixes to the filesystem (or
+    // - If `--fix` or `--fix-only` is set, apply applicable fixes to the filesystem (or
     //   print them to stdout, if we're reading from stdin).
-    // - If `--diff` or `--fix-only` are set, don't print any violations (only
-    //   fixes).
+    // - If `--diff` or `--fix-only` are set, don't print any violations (only applicable fixes)
+    // - By default, applicable fixes only include [`Applicablility::Automatic`], but if
+    //   `--fix--suggested` is set, then [`Applicablility::Suggested`] fixes are included.
+
+    let fix_suggested = if fix_suggested {
```

Comment:
I feel like passing `--fix-suggested` without any of `--fix`, `--diff`, etc. should raise an error on the CLI. Can we enforce that with Clap?